src/NLog/Internal/AssemblyHelpers.cs

```diff
@@ -102,9 +103,36 @@ public static Assembly LoadFromName(string assemblyName)
             Assembly assembly = assemblyPart.Load(stream.Stream);
             return assembly;
 #else
-            Assembly assembly = Assembly.Load(assemblyName);
-            return assembly;
+            try
+            {
+                Assembly assembly = Assembly.Load(AssemblyName.GetAssemblyName(assemblyName));
+                return assembly;
+            }
+            catch (FileNotFoundException)
+            {
+                var name = new AssemblyName(assemblyName);
+                var loadedAssembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(domainAssembly => CheckAssembly(name, domainAssembly.GetName()));
+                if (loadedAssembly != null)
+                    return loadedAssembly;
+                else throw;
+            }
+#endif
+        }
+
+        static bool CheckAssembly(AssemblyName expected, AssemblyName actual)
+        {
+            if (expected.Name != actual.Name)
+                return false;
+            if (expected.Version != null && expected.Version != actual.Version)
+                return false;
+#if !NETSTANDARD1_3 && !NETSTANDARD1_5
+            if (expected.CultureInfo != null && expected.CultureInfo.Name != actual.CultureInfo.Name)
+                return false;
 #endif
+            var expectedKeyToken = expected.GetPublicKeyToken();
+            if (expectedKeyToken != null && !expectedKeyToken.SequenceEqual(actual.GetPublicKeyToken()))
```

Comment:
Minor, I prefer a variable and return that variable. 

(Instead of if true, return true/false)

Reply:
That's a style, I don't like. :) But because it's your codebase I will change it. :)

Reply:
PS. http://wiki.c2.com/?ReturnBooleanEvaluations

Reply:
I usually prefer the `return a > b`-Style, but in that case it would have led to an expression like 
```
return !(expected.Name != actual.Name
    || (expected.Version != null && expected.Version != actual.Version)
    || (expected.CultureInfo != null && expected.CultureInfo.Name != actual.CultureInfo.Name)
    || (expected.GetPublicKeyToken() != null && !expected.GetPublicKeyToken().SequenceEqual(actual.GetPublicKeyToken())));
```
but in my opinion this is less explicit than my version.

Reply:
No I don't advocate that. ;)

Just no 

```C#
if(true) return true else return false
```

This is OK
```c#
if(true) return true;
somethingelse()
```

Reply:
See commit. Resolved now :)